trollflow2/launcher.py

```diff
@@ -54,26 +54,52 @@
 DEFAULT_PRIORITY = 999
 
 
-def run(prod_list, topics=None):
+def get_test_message(test_message_file):
+    """Read file and retrieve the test message"""
+
+    msg = None
+    if test_message_file:
+        with open(test_message_file) as fpt:
+            msg = fpt.readline().strip('\n')
+
+    return msg
+
+
+def run(prod_list, topics=None, test_message=None):
+
+    tmessage = get_test_message(test_message)
+    if tmessage:
+        from threading import Thread as Process
+        from posttroll.message import Message
+    else:
+        from multiprocessing import Process
 
     with open(prod_list) as fid:
         config = yaml.load(fid.read(), Loader=BaseLoader)
     topics = topics or config['product_list'].pop('subscribe_topics', None)
 
-    listener = ListenerContainer(topics=topics)
+    if not tmessage:
+        listener = ListenerContainer(topics=topics)
 
     while True:
         try:
-            msg = listener.output_queue.get(True, 5)
+            if tmessage:
+                mymessage = tmessage
```

Comment:
This is unnecessary, `tmessage` can be used directly without creating another variable.

Reply:
Right, thx, fixed.